### ISSUE TITLE
backpropagation of gradients into labels in tensorflow tf.nn.ctc_loss()

### ISSUE THREAD

**Issue by abhiramsingh0** [tensorflow/tensorflow#24728]
**System information**
- TensorFlow version (you are using): 1.12
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): Binary
- Python version: 2.7

**Describe the feature and the current behavior/state.**

I am working with tensorflow ctc loss function `tf.nn.ctc_loss(labels,inputs,__)`

Here `inputs` argument is the output from some RNN1 and `labels` argument is true target that we want to achieve.
In my case `labels` is also generated from some other RNN2.

But after training, only RNN1 weights are updated by tensorflow backpropagation and RNN2 weights are not changing at all. I think tensorflow does not support backpropagation of **gradients** into `labels` for **ctc_loss** as done in `tf.nn.softmax_cross_entropy_with_logits_v2()`.

What is the possible workaround for it?

**Will this change the current api? How?**

New parameter can be added to existing api `tf.nn.ctc_loss()` to enable or disable gradients backpropagation along `labels`.

**Who will benefit with this feature?**
A wast community of DL developers working with ctc_loss as labels can be generated dynamically from another neural network.

**Comment by vidursatija** [comment]
Hi,
By default CTC wasn't meant to backpropgrate through labels because they are the ground truth inputs. The reason why it worked with softmax cross entropy was because the loss got differentiated in the graph itself with respect to the labels. This isn't the case with ctc_loss as the labels don't have a gradient.

**Comment by abhiramsingh0** [comment]
> Hi,
> By default CTC wasn't meant to backpropgrate through labels because they are the ground truth inputs. The reason why it worked with softmax cross entropy was because the loss got differentiated in the graph itself with respect to the labels. This isn't the case with ctc_loss as the labels don't have a gradient.

Why softmax cross entropy loss was differentiated w.r.t ground truth labels?
If anyway differentiation was performed in cross entropy loss, similar thing can be achieved in ctc_loss.
I do not see any problem in that.

**Comment by vidursatija** [comment]
It might be possible. Someone will need to write the custom gradient implementation for it

**Comment by msymp** [comment]
Hello @abhiramsingh0 , @vidursatija has addressed the original query for this issue. This issue shall be closed. But feel free to raise a PR for gradient implementation for ctc_loss, if necessary. Thanks.
